R/evaluate.R

```diff
@@ -121,6 +121,11 @@ evaluate <- function(input,
   }
   local_inject_funs(envir)
 
+  if (is.null(getOption("rlang_trace_top_env"))) {
+    # If not already set, indicate the top environment to trim traceback
+    options(rlang_trace_top_env = envir)
+  }
```

Comment:
@hadley doing this in **evaluate** allows to get trimmed traceback in `evaluate()` itself. This is related to comment at https://github.com/r-lib/evaluate/pull/232#discussion_r1893974574
See the new snapshot file without any evaluate internals. 

For full context, this will be done at **knitr** level and so it will apply for `evaluate()` in **knitr** context. 
However, I don't know of the impact on doing that directly in **evaluate**. `evaluate()` is used also in other tools (like **testthat**) and I wonder if this could have impact. 

Maybe @lionel- you have some advice on setting this here ?

Reply:
My gut feeling is that it's worth it, but I don't know if there will be other negative effects.